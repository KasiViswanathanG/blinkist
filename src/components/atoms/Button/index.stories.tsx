// // import React from "react";
// // import Button from ".";

// // import { ThemeProvider } from '@mui/material/styles'
// // import CustomTheme from "../../../themes";

// // const ButtonComponent = {
// //   title: "Button",
// //   component: Button,
// //   argTypes: { onClick: { action: "clicked" } },
// // };

// // const Template = (args) => (
// //   <ThemeProvider theme={CustomTheme}>
// //     <Button {...args} />
// //   </ThemeProvider>
// // );

// // export const dynamicButton = Template.bind({});
// // dynamicButton.args = {
// //   variant:"contained",
// //   color:"primary",
// //   hoverColor : 'white',
// //   startIcon:null,
// //   endIcon:null,
// //   title:'Button',
// //   titleColor: 'black',
// //   buttonDeco: "none" ,
// //   width:130,
// //   borderColor:'black'
// // };

// // export default ButtonComponent;

// import React from 'react';
// import {Meta} from '@storybook/react';
// import ButtonComponent from '.';

// import { ThemeProvider } from '@mui/material/styles'
// import CustomTheme from "../../../themes";


// import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
// import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
// import AddIcon from '@mui/icons-material/Add';

// const meta: Meta = {
//   title: 'Button',
//   component: ButtonComponent
// }

// export default meta;

// export const outlined = () =>
//   <ThemeProvider theme={CustomTheme}>
//     <ButtonComponent variant={'outlined'} color={undefined} hoverColor={''} startIcon={''} endIcon={''} title={'Read now'} titleColor={"primary"} buttonDeco={'none'} width={100} borderColor={'#3A4649'} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
//   </ThemeProvider>

// export const outlinedWithIcon = () =>
// <ThemeProvider theme={CustomTheme}>
//   <ButtonComponent variant={'outlined'} color={undefined} hoverColor={''} startIcon={''} endIcon={<ArrowForwardIcon />} title={'Read now'} titleColor={"primary"} buttonDeco={'none'} width={150} borderColor={'#3A4649'} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
// </ThemeProvider>

// export const contained = () =>
//   <ThemeProvider theme={CustomTheme}>
//     <ButtonComponent variant={'contained'} color={'primary'} hoverColor={'#20BA68'} startIcon={''} endIcon={''} title={'Finished Reading'} titleColor={undefined} buttonDeco={'none'} width={150} borderColor={''} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
//   </ThemeProvider>

// export const containedWithIcon = () =>
// <ThemeProvider theme={CustomTheme}>
//   <ButtonComponent variant={'contained'} color={'primary'} hoverColor={'#20BA68'} startIcon={''} endIcon={<ArrowForwardIcon />} title={'Finished Reading'} titleColor={undefined} buttonDeco={'none'} width={200} borderColor={''} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
// </ThemeProvider>

// export const containedWithIconSecondary = () =>
// <ThemeProvider theme={CustomTheme}>
//   <ButtonComponent variant={'contained'} color={'secondary'} hoverColor={''} startIcon={<AddIcon />} endIcon={''} title={'Add To Library'} titleColor={"inherit"} buttonDeco={'none'} width={284} borderColor={'#E5E4E2'} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
// </ThemeProvider>

// export const text = () =>
//   <ThemeProvider theme={CustomTheme}>
//     <ButtonComponent variant={'text'} color={undefined} startIcon={''} endIcon={''} title={'My Library'} titleColor={'black'} buttonDeco={'none'} hoverColor={''} width={91} borderColor={''} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''}  />
//   </ThemeProvider>

// export const textOnHover = () =>
//   <ThemeProvider theme={CustomTheme}>
//     <ButtonComponent variant={'text'} color={undefined} startIcon={''} endIcon={''} title={'My Library'} titleColor={'black'} buttonDeco={'none'} hoverColor={''} width={91} borderColor={''} onClick={() => { } } borderBottom={'2px solid'} paddingBottom={'1px'} paddingTop={'3px'} backgroundColor={'white'}  />
//   </ThemeProvider>

// export const textWithIcon = () =>
//   <ThemeProvider theme={CustomTheme}>
//     <ButtonComponent variant={'text'} color={undefined} startIcon={''} endIcon={<KeyboardArrowDownIcon htmlColor='black' />} title={'Explore'} titleColor={'black'} buttonDeco={'none'} hoverColor={''} width={90} borderColor={''} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''}/>
//   </ThemeProvider>

// export const textWithIconSecondary = () =>
// <ThemeProvider theme={CustomTheme}>
//   <ButtonComponent variant={'text'} color={'secondary'} hoverColor={''} startIcon={<AddIcon />} endIcon={''} title={'Add To Library'} titleColor={"inherit"} buttonDeco={'none'} width={284} borderColor={'#E5E4E2'} onClick={() => { } } borderBottom={''} paddingBottom={''} paddingTop={''} backgroundColor={''} />
// </ThemeProvider>

// import React from 'react'

// function sto() {
//   return (
//     <div>
      
//     </div>
//   )
// }

// export default sto

import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Button from '.';
import CustomTheme from "../../../themes";
import { ThemeProvider } from '@mui/material/styles'

import { addDecorator } from '@storybook/react';

import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import AddIcon from '@mui/icons-material/Add';
import { Typography } from '@mui/material';

export default {
  title: 'Atoms/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

addDecorator((story) => (
  <ThemeProvider theme={CustomTheme}>{story()}</ThemeProvider>
));

export const Default = Template.bind({});
Default.args = {
  children: 'Button',
};

export const Contained = Template.bind({});
Contained.args = {
  variant: 'contained',
  color: 'primary',
  sx: { '&:hover': { backgroundColor: '#20BA68' } },
  children: 'Finished Reading',
};

export const ContainedWithIcon = Template.bind({});
ContainedWithIcon.args = {
  variant: 'contained',
  color: 'primary',
  endIcon: <ArrowForwardIcon />,
  sx: { '&:hover': { backgroundColor: '#20BA68' } },
  children: 'Finished Reading',
};

export const ContainedWithIconSecondary = Template.bind({});
ContainedWithIconSecondary.args = {
  variant: 'contained',
  color: 'secondary',
  startIcon: <AddIcon />,
  sx: {width:284 },
  children: 'Add To Library',
};

export const Outlined = Template.bind({});
Outlined.args = {
  variant: 'outlined',
  children: 'Read now',
};

export const OutlinedWithIcon = Template.bind({});
OutlinedWithIcon.args = {
  variant: 'outlined',
  endIcon: <ArrowForwardIcon />,
  children: 'Read now',
};

export const Disabled = Template.bind({});
Disabled.args = {
  variant: 'contained',
  disabled: true,
  children: 'Button',
};

export const Text = Template.bind({});
Text.args = {
  variant: 'text',
  children: <Typography color='black'>My Library</Typography>,
};

export const TextOnHover = Template.bind({});
TextOnHover.args = {
  variant: 'text',
  sx: {'&:hover': {
    borderBottom : '2px solid',
    backgroundColor: 'white'},
    borderRadius:'0px'},
  children: <Typography color='black'>My Library</Typography>,
};

export const TextWithIcon = Template.bind({});
TextWithIcon.args = {
  variant: 'text',
  endIcon: <KeyboardArrowDownIcon htmlColor='black' />,
  sx: {'&:hover': {
    borderBottom : '2px solid',
    backgroundColor: 'white'},
    borderRadius:'0px'},
  children: <Typography color='black'>Explore</Typography>,
};

export const TextWithIconSecondary = Template.bind({});
TextWithIconSecondary.args = {
  variant: 'text',
  color: 'secondary',
  startIcon: <AddIcon />,
  sx: {width:'284'},
    children: 'Add To Library',
};