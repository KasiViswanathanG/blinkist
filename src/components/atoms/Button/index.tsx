// import { Button, Typography } from '@mui/material'
// import React from 'react'

// type ButtonProp = {
//   variant:"text" | "outlined" | "contained" | undefined
//   color:"inherit" | "primary" | "secondary" | "success" | "error" | "info" | "warning" | undefined
//   hoverColor : string
//   startIcon:React.ReactNode | null
//   endIcon:React.ReactNode | null
//   title:string
//   titleColor: string | undefined
//   buttonDeco: "none" | "underline"
//   width:number
//   borderColor:string
//   onClick:(event: React.MouseEvent<HTMLButtonElement>) => void
//   borderBottom:string
//   paddingBottom:string
//   paddingTop:string
//   backgroundColor:string
// }

// const ButtonComponent = ({variant,color,hoverColor,startIcon,endIcon,title,titleColor,buttonDeco,width,borderColor,onClick,borderBottom,paddingBottom,paddingTop,backgroundColor}: ButtonProp) => {
//   return (
//     <Button 
//     variant={variant}
//     color={color}
//     startIcon={startIcon}
//     endIcon={endIcon}
//     onClick={onClick}
//     sx={{'&:hover':{backgroundColor: hoverColor},textDecoration: buttonDeco, width:width, borderColor:borderColor,
//     borderBottom : {borderBottom},
//     paddingBottom : {paddingBottom},
//     paddingTop : {paddingTop},
//     backgroundColor: {backgroundColor}}}>
//       <Typography color={titleColor}>
//         {title}
//       </Typography>
//     </Button>
//   )
// }

// export default ButtonComponent


import { styled } from '@mui/material/styles';
import { Button as MUIButton, ButtonProps } from '@mui/material';

import CustomTheme from "../../../themes";

const StyledButton = styled(MUIButton)({
  ...CustomTheme.typography.body1,
  textTransform: 'none',
  borderRadius: '4px',
});

interface IButtonProps extends ButtonProps {}

const Button = ({ children, ...rest }: IButtonProps) => {
  return (
    <StyledButton {...rest} data-testid={children}>
      {children}
    </StyledButton>
  );
};

export default Button;
