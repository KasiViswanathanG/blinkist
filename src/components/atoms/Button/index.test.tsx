import {screen, fireEvent, getByText, render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import Button from '.'
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
import AddIcon from '@mui/icons-material/Add';

describe('Testing the Button', () => {
  it('render default button', () => {
    const button = renderer.create(<Button>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render disabled button', () => {
    const button = renderer.create(<Button disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render contained button', () => {
    const button = renderer.create(<Button variant='contained' disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render contained button with icon', () => {
    const button = renderer.create(<Button variant='contained' endIcon={<ArrowForwardIcon />} disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render outlined button', () => {
    const button = renderer.create(<Button variant='outlined' disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render outlined button with icon', () => {
    const button = renderer.create(<Button variant='outlined' endIcon={<ArrowForwardIcon />} disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render text button', () => {
    const button = renderer.create(<Button variant='text' disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render text button with icon', () => {
    const button = renderer.create(<Button variant='text' startIcon={<AddIcon />} disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('render secondary button', () => {
    const button = renderer.create(<Button variant='text' color='secondary' disabled>Button</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('should call on click', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Finished</Button>)
    const buttonElement = screen.getByText('Finished')
    fireEvent.click(buttonElement)
    expect(handleClick).toHaveBeenCalled();
  });
})