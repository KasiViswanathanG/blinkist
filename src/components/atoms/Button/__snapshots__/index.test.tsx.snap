// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Button render contained button 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-1og0dbc-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render contained button with icon 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-1og0dbc-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
  <span
    className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowForwardIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Testing the Button render default Button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render default button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render disabled Button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render disabled button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render outlined button 1`] = `
<button
  className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled css-f9ruw4-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render outlined button with icon 1`] = `
<button
  className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled css-f9ruw4-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
  <span
    className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowForwardIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Testing the Button render secondary button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-x5bmov-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render text button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  Button
</button>
`;

exports[`Testing the Button render text button with icon 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled css-u0ari6-MuiButtonBase-root-MuiButton-root"
  data-testid="Button"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </span>
  Button
</button>
`;
